test/python/transpiler/test_transpile.py

```diff
@@ -5,7 +5,7 @@
 # This source code is licensed under the Apache License, Version 2.0 found in
 # the LICENSE.txt file in the root directory of this source tree.
 
-# pylint: disable=redefined-builtin
+# pylint: disable=redefined-builtin,invalid-name
```

Comment:
Can you just use a valid name here instead of disabling pylint? There was an effort a while ago (https://github.com/Qiskit/qiskit-terra/pull/739) to minimize these ignores, so we should try to not reintroduce them. If the pylint is too restrictive with names, we should try to relax that.

Reply:
Pylint is overly pedantic and this rule is just an example of that. The rule here is set to check all function names conform to this regex: `(([a-z_][a-z0-9_]{2,49})|(assert[A-Z][a-zA-Z0-9]{2,43}))$` see: https://github.com/Qiskit/qiskit-terra/blob/master/.pylintrc#L160 . In the case of this patch it fails because the function name is 54 characters instead of between 2 and 49.  Which is a rule quite clearly written for non-test code assuming the function will be called (the character limit is for overall line length), but with test code the function name is a unique identifier for the test and is never called directly. You want a test function name to be as descriptive as possible so you can identify what it does. Blindly applying the pylint rules just because we can doesn't really provide us any benefits and just cause headaches. Honestly we need to reduce our pylint rule set to a more manageable set, or change to another tool (see #1179 ).

Reply:
Maybe extend the rule to
`method-rgx=(([a-z_][a-z0-9_]{2,49})|(assert[A-Z][a-zA-Z0-9]{2,43})|(test_[a-zA-Z0-9]{2}))$` ?

Reply:
That works for me, it's the same net effect of disabling the rule for test code. I'll push an update adding that and removing this line. It does raise the meta-question about this, do people actually find value in enforcing a method name length?